syntax = "proto3";

package tbk_log;

// 定义不同的消息类型
enum MessageType {
  MESSAGE_JSON = 0;
  MESSAGE_PROTO = 1;
}

// 定义日志文件头部
message LogFileHeader {
  string file_type = 1; // 文件类型
  int32 format_version = 2; // 格式版本
  string checksum = 3; // 文件校验和，用于验证文件完整性
}

// 定义日志消息
message LogMessage {
  int64 timestamp = 1; // 接收时间戳，单位为纳秒
  MessageType message_type = 2; // 消息类型
  int32 message_size = 3; // 二进制 protobuf 消息的大小
  bytes message_data = 4; // 二进制 protobuf 消息
  int32 version = 5; // 消息版本，用于向后兼容
  string json_data = 6; // JSON 消息数据（如果 message_type 是 MESSAGE_JSON）
}


// 定义完整的日志文件结构
message LogFile {
  LogFileHeader header = 1; // 文件头部
  repeated LogMessage messages = 2; // 日志消息

}